[CI] EsqlSpecIT test {csv-spec:subquery.SubqueryInFromWithRerankInSubquery} failing

**Build Scans:**
- [elasticsearch-intake #34686 / part3](https://gradle-enterprise.elastic.co/s/26ydgsyn7ii72)
- [elasticsearch-pull-request #120086 / part-3](https://gradle-enterprise.elastic.co/s/kxfy6bdszq5ms)
- [elasticsearch-pull-request #120066 / part-3](https://gradle-enterprise.elastic.co/s/wq7bj37idvje6)
- [elasticsearch-pull-request #120050 / part-3](https://gradle-enterprise.elastic.co/s/qzm5qumiijebw)
- [elasticsearch-pull-request #120007 / part-3](https://gradle-enterprise.elastic.co/s/nna2yq25qvmfm)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:multi-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT" -Dtests.method="test {csv-spec:subquery.SubqueryInFromWithRerankInSubquery}" -Dtests.seed=F48A62308BFA920E -Dtests.locale=dyo -Dtests.timezone=Europe/Lisbon -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Bcsv-spec%3Asubquery.SubqueryInFromWithRerankInSubquery%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: unexpected partial results: _clusters={details={(local)={_shards={total=3, failed=2, successful=1, skipped=0}, took=55, indices=books,sample_data, failures=[{node=null, reason={reason=Index 1 out of bounds for length 0, type=array_index_out_of_bounds_exception}, index=books, shard=2}], status=successful}}}
Expected: (null or is <false>)
     but: was <true>
```

**Issue Reasons:**
- [main] 5 failures in test test {csv-spec:subquery.SubqueryInFromWithRerankInSubquery} (0.5% fail rate in 936 executions)
- [main] 4 failures in step part-3 (0.7% fail rate in 575 executions)
- [main] 4 failures in pipeline elasticsearch-pull-request (0.7% fail rate in 570 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -419,9 +419,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT
   method: test {p0=esql/40_tsdb/TS Command grouping on text field}
   issue: https://github.com/elastic/elasticsearch/issues/142544
-- class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
-  method: test {csv-spec:subquery.SubqueryInFromWithRerankInSubquery}
-  issue: https://github.com/elastic/elasticsearch/issues/142552
 - class: org.elasticsearch.index.IndexSettingsTests
   method: testSyntheticIdBadVersion
   issue: https://github.com/elastic/elasticsearch/issues/142508

